sched/sched/sched_unlock.c

```diff
@@ -37,281 +37,178 @@
 #include "sched/sched.h"
 
 /****************************************************************************
- * Public Functions
+ * Private Data
  ****************************************************************************/
 
+#if defined(CONFIG_SCHED_TICKLESS) && \
+           (CONFIG_RR_INTERVAL > 0 || defined(CONFIG_SCHED_SPORADIC))
+static int preempt_schedule_nestcount = 0;
```

Comment:
could we remove this global variable?

Reply:
nxsched_reassess_timer may also trigger sched_preempt_schedule

Reply:
The logic before optimization does not have this issue? do you have the backtrace of recursive call?

Reply:
Because we have placed sched_lock inside spin_lock_irqsave,
and spin_lock_irqsave is also extensively used to replace critical sections,
the usage of sched_lock has become very widespread, which makes recursion highly likely to occur.

Reply:
ok, how about decrease lockcount after call `sched_preemt_schedule()`, This will ensure that the unlock operation is not recursively processed in context switch flow.
![image](https://github.com/user-attachments/assets/ef5f55c2-64ca-4232-9d34-e8fc60b1e0ef)

Reply:
Due to the scheduling logic in sched_preemt_schedule that relies on checking the lockcount to determine the state,
if the lockcount is still greater than 0, we cannot distinguish whether the schedlock has been released or not.

Reply:
or should we move this variable into tcb? I don't agree a global variable constantly updated in the scheduler, this is bad design.

Reply:
tcb->flags should be able to solve this problem.